File: src/main/java/com/kajipoi/entity/Work.java
package com.kajipoi.entity;

import com.kajipoi.domain.Id;
import com.kajipoi.domain.Status;
import lombok.Data;
import org.seasar.doma.Entity;
import org.seasar.doma.jdbc.entity.NamingType;

import java.time.LocalDate;

/**
 * 仕事イベントを表すエンティティ。
 *
 * @author syobochim
 * @since 1.0
 */
@Entity(naming = NamingType.LOWER_CASE)
@Data
public class Work {
    private Id memberId;
    private Id houseworkId;
    private LocalDate workData;

    private Status status;
}

File: src/main/kotlin/kajipoi/entityJava/Work.kt
package kajipoi.entityJava

import kajipoi.domain.Id
import kajipoi.domain.Status
import org.seasar.doma.Entity
import org.seasar.doma.jdbc.entity.NamingType

/**
 * @author syobochim
 */
@Entity(naming = NamingType.LOWER_CASE, immutable = true)
class Work(val memberId: Id,
           val houseId: Id,
           val workData: Id,
           val status: Status)